pkg/broker/deprovision_subscriber.go

```diff
@@ -66,8 +71,11 @@ func setFailedDeprovisionJob(dao SubscriberDAO, dmsg JobMsg) {
 }
 
 func cleanupDeprovision(instanceID string, dao SubscriberDAO) error {
-	if err := dao.DeleteExtractedCredentials(instanceID); err != nil {
-		log.Error("failed to delete extracted credentials - %v", err)
+	err := dao.DeleteExtractedCredentials(instanceID)
+
+	if err != nil && strings.Contains(err.Error(), etcdKeyNotFoundContents) {
```

Comment:
All the other places we do `dao` calls this is how we check for the not found key error:

```
    if err != nil {
        // no jobs or states to recover, this is OK.
        if client.IsKeyNotFound(err) {
            log.Info("No jobs to recover")
            return "", nil
        }
        return "", err
    }
```

The above comes from `broker.go`